### PR TITLE
[Feature] show a loading bar when creating a post

### PR DESCRIPTION
**Pull request by Xiawpohr**
## Summary

文章點擊送出後，可以蓋一個loading bar或是disable取消編輯的按鈕。

## Linked Issue

[#104](https://github.com/social-tw/social-tw-website/issues/104)

## Impacted Areas

- packages/frontend/src/components/post/PostForm.tsx
- packages/frontend/src/hooks/useCreatePost.ts
- packages/frontend/src/pages/PostCreate.tsx
- packages/frontend/src/pages/PostList.tsx
- packages/frontend/src/utils/makeCancellableTask.ts

## Todo

Break down the progress of the PR for everyone to see what else you intend to include in this PR.

-  [x] make create process cancellable
-  [x] add cancel submitting button
-  [x] show pending bar after publishing a post

### CODE REVIEW

**Comment by kittybest** [comment]
image

progress bar會一直蓋在上面，上圖為註冊還未完成的時候。

image

上圖為已完成註冊但無法po文。

image

手機版有一樣的問題。

File: packages/frontend/src/components/post/PostForm.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                                        'min-h-[3rem] overflow-auto text-white text-xl',
+                                    placeholder: 'text-gray-300 text-xl',
+                                }}
+                            />
+                        )}
+                    />
+                </section>
+            </form>
+            <Modal
+                isOpen={isPending || true}
```
Comment (kittybest) [inline packages/frontend/src/components/post/PostForm.tsx:95]: 是因為還在測試階段嗎？
因為這邊是true所以無論如何都會打開這個Modal。

File: packages/frontend/src/pages/PostCreate.tsx [hunk 2]
```diff
@@ -14,24 +15,35 @@ export default function PostCreate() {
```
Comment (kittybest) [inline packages/frontend/src/pages/PostCreate.tsx:15]: `const {isLogin}` is not being used, could be removed.

File: packages/frontend/src/components/post/PostForm.tsx (hunk 1 above)
Comment (Xiawpohr) [reply packages/frontend/src/components/post/PostForm.tsx:95]: 抱歉，我忘了把 true 刪掉了

**Review by kittybest — APPROVED** [review]

**Pull request merged**